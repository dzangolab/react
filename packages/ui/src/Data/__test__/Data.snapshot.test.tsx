import { render } from "@testing-library/react";
import React from "react";
import { expect, test, describe } from "vitest";

import { Data } from "..";

describe("Data Component", () => {
  test("Component matches snapshot with all props", () => {
    const { container } = render(<Data caption="Name" value="John" />);
    expect(container).toMatchSnapshot();
  });

  test("matches the snapshot with 'stat' mode", () => {
    const { container } = render(
      <Data caption="Name" value="John" mode="stat" />,
    );
    expect(container).toMatchSnapshot();
  });

  test("matches the snapshot with custom className", () => {
    const { container } = render(
      <Data caption="Name" value="John" className="custom-class" />,
    );
    expect(container).toMatchSnapshot();
  });

  test("matches the snapshot with ReactNode as caption and value", () => {
    const caption = <span>Name</span>;
    const value = <strong>John</strong>;

    const { container } = render(<Data caption={caption} value={value} />);
    expect(container).toMatchSnapshot();
  });
});
