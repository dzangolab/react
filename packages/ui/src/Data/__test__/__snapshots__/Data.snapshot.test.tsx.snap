// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Data Component > Component matches snapshot with all props 1`] = `
<div>
  <div
    class="data"
  >
    <div
      class="label"
    >
      Name
    </div>
    <div
      class="value"
    >
      John
    </div>
  </div>
</div>
`;

exports[`Data Component > matches the snapshot with 'stat' mode 1`] = `
<div>
  <div
    class="data data-stat"
  >
    <div
      class="label"
    >
      Name
    </div>
    <div
      class="value"
    >
      John
    </div>
  </div>
</div>
`;

exports[`Data Component > matches the snapshot with ReactNode as caption and value 1`] = `
<div>
  <div
    class="data"
  >
    <div
      class="label"
    >
      <span>
        Name
      </span>
    </div>
    <div
      class="value"
    >
      <strong>
        John
      </strong>
    </div>
  </div>
</div>
`;

exports[`Data Component > matches the snapshot with custom className 1`] = `
<div>
  <div
    class="data  custom-class"
  >
    <div
      class="label"
    >
      Name
    </div>
    <div
      class="value"
    >
      John
    </div>
  </div>
</div>
`;
